amy-message-hub/src/main/java/de/unistuttgart/iaas/amyassist/amy/messagehub/MessageHubService.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * This source file is part of the Amy open source project.
+ * For more information see github.com/AmyAssist
+ * 
+ * Copyright (c) 2018 the Amy project authors.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information see notice.md
+ */
+
+package de.unistuttgart.iaas.amyassist.amy.messagehub;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.function.Consumer;
+
+import org.slf4j.Logger;
+
+import de.unistuttgart.iaas.amyassist.amy.core.di.annotation.PostConstruct;
+import de.unistuttgart.iaas.amyassist.amy.core.di.annotation.Reference;
+import de.unistuttgart.iaas.amyassist.amy.core.di.annotation.Service;
+
+/**
+ * 
+ * @author Kai Menzel, Leon Kiefer
+ */
+@Service
+public class MessageHubService implements MessageHub, Runnable {
+
+	@Reference
+	private Logger logger;
+
+	private Map<String, List<Consumer<String>>> eventListener = new HashMap<>();
+
+	private BlockingQueue<Message<String>> queue = new LinkedBlockingQueue<>();
+
+	private Thread thread;
+
+	@PostConstruct
+	private void start() {
+		this.thread = new Thread(this, "MessageHub");
+		this.thread.start();
+	}
+
+	@Override
+	public void run() {
+		while (!Thread.currentThread().isInterrupted()) {
+			try {
+				Message<String> event = this.queue.take();
+				if (this.eventListener.containsKey(event.getTopic())) {
+					List<Consumer<String>> list = this.eventListener.get(event.getTopic());
+					for (Consumer<String> eventExecuter : list) {
+						executeHandler(eventExecuter, event);
+					}
+				}
+			} catch (InterruptedException e) {
```

Comment:
Should this not be logged?

Reply:
no, because this is the normal control flow if the the MessageHub is stopped

Reply:
How would a scenario, where MessageHub is stopped, look like?

Reply:
you type `exit` in the console